pilot/pkg/kube/inject/app_probe.go

```diff
@@ -123,7 +139,16 @@ func rewriteAppHTTPProbe(spec *SidecarInjectionSpec, podSpec *corev1.PodSpec) {
 		for _, p := range c.Ports {
 			portMap[p.Name] = p.ContainerPort
 		}
-		rewriteProbe(c.ReadinessProbe, portMap)
-		rewriteProbe(c.LivenessProbe, portMap)
+		rewriteProbe(c.ReadinessProbe, &appProberInfo, fmt.Sprintf("/app-health/%v/readyz", c.Name), statusPort, portMap)
+		rewriteProbe(c.LivenessProbe, &appProberInfo, fmt.Sprintf("/app-health/%v/livez", c.Name), statusPort, portMap)
+	}
+
+	// Finally propagate app prober config to `istio-proxy` through command line flag.
+	b, err := json.Marshal(appProberInfo)
```

Comment:
ah, see this is auto injected.

Reply:
yup, and the flag is documented with "never should be set manually"

Reply:
There is an issue here with new injector webhook working with with older version of proxy image. Older proxy image does not understand this flag, it will error out.
The webhook needs to look at the version of proxy to know if it is safe to inject this param.

The alternative is to use and environment variable.